evm_arithmetization/src/verifier.rs

```diff
@@ -240,12 +256,39 @@ pub mod testing {
             .map(|i| get_memory_extra_looking_sum(&public_values, ctl_challenges.challenges[i]))
             .collect_vec();
 
+        let all_ctls = &all_stark.cross_table_lookups;
+
         verify_cross_table_lookups::<F, D, NUM_TABLES>(
             cross_table_lookups,
-            all_proof
-                .multi_proof
-                .stark_proofs
-                .map(|p| p.proof.openings.ctl_zs_first.unwrap()),
+            core::array::from_fn(|i| {
+                if let Some(stark_proof) = &stark_proofs[i] {
+                    stark_proof
+                        .proof
+                        .openings
+                        .ctl_zs_first
+                        .as_ref()
+                        .expect("Missing ctl_zs")
+                        .clone()
+                } else if i == *Table::Keccak {
```

Comment:
I assume this will get refactored in the generalization to other tables as well?

Reply:
Yes. It got refactored in #724